src/EventListener/RequestTokenListener.php

```diff
@@ -0,0 +1,126 @@
+<?php
+
+/*
+ * This file is part of Contao.
+ *
+ * Copyright (c) 2005-2015 Leo Feyer
+ *
+ * @license LGPL-3.0+
+ */
+
+
+namespace Contao\CoreBundle\EventListener;
+
+use Contao\CoreBundle\Exception\AjaxRedirectResponseException;
+use Contao\CoreBundle\Exception\InvalidRequestTokenException;
+use Contao\CoreBundle\Framework\ContaoFrameworkInterface;
+use Symfony\Component\HttpKernel\Event\GetResponseEvent;
+use Symfony\Component\Routing\RouterInterface;
+use Symfony\Component\Security\Csrf\CsrfToken;
+use Symfony\Component\Security\Csrf\CsrfTokenManagerInterface;
+
+/**
+ * Validates Contao request tokens.
+ *
+ * @author Yanick Witschi <https://github.com/toflar>
+ */
+class RequestTokenListener
+{
+    /**
+     * @var ContaoFrameworkInterface
+     */
+    private $framework;
+
+    /**
+     * @var array
+     */
+    private $routeNames;
+
+    /**
+     * @var RouterInterface
+     */
+    private $router;
+
+    /**
+     * @var CsrfTokenManagerInterface
+     */
+    private $tokenManager;
+
+    /**
+     * @var string
+     */
+    private $csrfTokenName;
+
+    /**
+     * Constructor.
+     *
+     * @param ContaoFrameworkInterface  $framework
+     * @param array                     $routeNames
+     * @param RouterInterface           $router
+     * @param CsrfTokenManagerInterface $tokenManager
+     * @param string                    $csrfTokenName
+     */
+    public function __construct(
+        ContaoFrameworkInterface $framework,
+        array $routeNames,
+        RouterInterface $router,
+        CsrfTokenManagerInterface $tokenManager,
+        $csrfTokenName
+    ) {
+        $this->framework = $framework;
+        $this->routeNames = $routeNames;
+        $this->router = $router;
+        $this->tokenManager = $tokenManager;
+        $this->csrfTokenName = $csrfTokenName;
+    }
+
+    /**
+     * Handle the request token.
+     *
+     * @param GetResponseEvent $event
+     *
+     * @throws AjaxRedirectResponseException|InvalidRequestTokenException If the token is invalid
+     */
+    public function onKernelRequest(GetResponseEvent $event)
+    {
+        if (!$this->framework->isInitialized()) {
+            return;
+        }
+
+        $request = $event->getRequest();
+
+        if (null === $request
+            || 'POST' !== $request->getRealMethod()
```

Comment:
Maybe we should use [`Request::isMethodSafe()`](https://github.com/symfony/symfony/blob/2.8/src/Symfony/Component/HttpFoundation/Request.php#L1468) so we also protect `PUT` etc.?
If not we should at least use `Request::isMethod('POST')`